File: sekre-mobile/composeApp/src/commonMain/kotlin/org/sekre_mobile/com/domain/usecase/task/UpdateTaskStatusUseCase.kt
package org.sekre_mobile.com.domain.usecase.task

import org.sekre_mobile.com.domain.entity.TaskStatus
import org.sekre_mobile.com.domain.model.Result
import org.sekre_mobile.com.domain.repository.TaskRepository

/**
 * Update Task Status Use Case
 * Application layer - orchestrates business logic
 */
class UpdateTaskStatusUseCase(
    private val taskRepository: TaskRepository
) {
    suspend operator fun invoke(
        id: String,
        status: TaskStatus
    ): Result<Unit> {
        if (id.isBlank()) {
            return Result.Error(Exception("Task ID is required"))
        }

        return taskRepository.updateTaskStatus(id, status)
    }
}

File: sekre-mobile/src/core/usecases/tasks/UpdateTaskStatusUseCase.ts
import type { ITaskRepository } from '@core/ports/ITaskRepository';
import type { Task, TaskId, TaskStatus } from '@core/domain/entities/Task';

export class UpdateTaskStatusUseCase {
  constructor(private readonly taskRepository: ITaskRepository) {}

  async execute(id: TaskId, status: TaskStatus): Promise<Task> {
    return this.taskRepository.updateTaskStatus(id, status);
  }
}
